theme/color.go

```diff
@@ -0,0 +1,55 @@
+package theme
+
+import (
+	"image/color"
+)
+
+// Brighten takes a color and brightens / dims it by the given
+// amount.
+// A value of 1.0 == the same
+// 1.1 = 10% brighter
+// 0.9 = 10% dimmer, etc
+func Brighten(c color.Color, amount float64) color.Color {
+	r, g, b, a := c.RGBA()
+	if IsDark() {
+		amount = amount - 2.0*(amount - 1.0)
+	}
+	fn := func(i uint32) uint8 {
+		x := uint32(float64(i) * amount)
+		if amount > 1.0 && x > 0xffff {
+			x = 0xffff
+		}
+		if amount < 1.0 && x > i {
+			x = 0
+		}
+		return uint8(x >> 8)
+	}
+	return color.RGBA{fn(r), fn(g), fn(b), uint8(a)}
+}
+
+// PrimaryTextColor returns either {black, white} depending on
+// the brightness of the primary color
+func PrimaryTextColor() color.Color {
```

Comment:
This is only used in the widget package, perhaps it can be a private fuction there instead?